matrix/manager.go

```diff
@@ -174,12 +200,34 @@ func (k *KubeManager) probeConnectivity(nsFrom, podFrom, containerFrom, addrTo s
 	commandDebugString := fmt.Sprintf("kubectl exec %s -c %s -n %s -- %s", podFrom, containerFrom, nsFrom, strings.Join(cmd, " "))
 	stdout, stderr, err := k.executeRemoteCommand(nsFrom, podFrom, containerFrom, cmd)
 	if err != nil {
-		fmt.Println(fmt.Printf("%s/%s -> %s: error when running command: err - %v /// stdout - %s /// stderr - %s", nsFrom, podFrom, addrTo, err, stdout, stderr))
+		fmt.Println(fmt.Printf("%s/%s -> %s: error when running command:"+
+			" err - %v /// stdout - %s /// stderr - %s", nsFrom, podFrom, addrTo, err, stdout, stderr))
 		return false, commandDebugString, nil
 	}
 	return true, commandDebugString, nil
 }
 
+// ProbeConnectivityWithCurl execs into a pod and connect the endpoint, return endpoint
+func (k *KubeManager) ProbeConnectivityWithCurl(nsFrom, podFrom, containerFrom, addrTo string, protocol v1.Protocol, toPort int) (bool, string, string, error) { // nolint
+	port := strconv.Itoa(toPort)
+
+	cmd := []string{
+		"/usr/bin/curl", "--connect-timeout", "5", "-g", "-q", "-s",
+		"--max-time", "10", "--retry", "5", "--retry-delay", "0", "--retry-max-time", "20", "telnet://" + net.JoinHostPort(addrTo, port),
```

Comment:
Why not use `agnhost connect`? are we missing parameters there?

Reply:
if the only difference is the output, why not reusing `ProbeConnectivity` and return from there.

Reply:
agnhost connect seems does not get back pod names as response. need to explore.

Reply:
have replaced with nc